File: packages/backend/src/main/java/com/keyVault/app/entity/Password.java
package com.keyVault.app.entity;
import java.util.Date;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty.Access;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
@Entity
@Table(name = "passwords")
public class Password {
	@Id
	@GeneratedValue(strategy =  GenerationType.IDENTITY)
	private int id;
	@Column(nullable = false)
	private String content;
	
	private String name;
	@Column(nullable = false)
	private String userOrMail;
	@JsonProperty(access = Access.WRITE_ONLY)
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "category_id")
	private Category category;
	@ManyToOne(fetch = FetchType.LAZY )
	@JoinColumn(name= "icon_id")
	private Icon icon;
	
	@ManyToOne(fetch = FetchType.LAZY )
	@JoinColumn(name= "user_id")
	private User user;
	
	private Date createdAt;
	
	private int seenqty;//number of times password seen
	
	private String url;

	public Password() {
		super();
	}
	
	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getContent() {
		return content;
	}

	public void setContent(String content) {
		this.content = content;
	}

	public Date getCreatedAt() {
		return createdAt;
	}

	public void setCreatedAt(Date createdAt) {
		this.createdAt = createdAt;
	}

	public int getSeenqty() {
		return seenqty;
	}

	public void setSeenqty(int seenqty) {
		this.seenqty = seenqty;
	}

	public Category getCategory() {
		return category;
	}

	public void setCategory(Category category) {
		this.category = category;
	}

	public Icon getIcon() {
		return icon;
	}

	public void setIcon(Icon icon) {
		this.icon = icon;
	}

	public String getUrl() {
		return url;
	}

	public void setUrl(String url) {
		this.url = url;
	}

	public String getUserOrMail() {
		return userOrMail;
	}

	public void setUserOrMail(String userOrMail) {
		this.userOrMail = userOrMail;
	}
	
}

File: packages/mobile/key_vault/lib/models/password.dart
import 'dart:convert';
import 'package:key_vault/models/models.dart';

Password passwordFromJson(String str) => Password.fromJson(json.decode(str));

String passwordToJson(Password data) => json.encode(data.toJson());

class Password {
  Password({
    required this.id,
    required this.content,
    required this.name,
    this.userOrMail,
    required this.createdAt,
    required this.seenQty,
    this.category,
    this.icon,
    this.url,
  });

  int id;
  String? content;
  String createdAt;
  int seenQty;
  String name;
  String? userOrMail;
  Category? category;
  Icon? icon;
  String? url;

  factory Password.fromJson(Map<String, dynamic> json) => Password(
        id: json["id"],
        content: json["content"],
        name: json["name"] ?? "No-Name",
        userOrMail: json["userOrMail"],
        createdAt: json["createdAt"],
        seenQty: json["seenqty"],
        category: json["category"] != null
            ? Category.fromJson(json["category"])
            : null,
        icon: json["icon"] != null ? Icon.fromJson(json["icon"]) : null,
        url: json["url"],
      );

  Map<String, dynamic> toJson() => {
        "id": id,
        "content": content,
        "name": name,
        "userOrMail": userOrMail,
        "createdAt": createdAt,
        "seenQty": seenQty,
        "category": category?.toJson(),
        "icon": icon?.toJson(),
        "url": url,
      };
}
